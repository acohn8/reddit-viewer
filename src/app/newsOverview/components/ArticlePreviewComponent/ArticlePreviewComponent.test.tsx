import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';

import ArticlePreview from './ArticlePreviewComponent';

describe('ArticlePreviewComponent', () => {
  const output = shallow(<ArticlePreview
    link="www.newyorker.com"
    source="New Yorker"
    text="a cool article2"
    title="breaking"
    image="image.jpg"
    postDate={new Date(1536423624 * 1000)}
    icon="newspaper outline"
    upVotes={7}
    comments={5}
  />);
  it('should render correctly', () => {
    expect(shallowToJson(output)).toMatchSnapshot();
  });
  it('should have an source', () => {
    expect(output.find('New Yorker'));
  });
  it('should have an image', () => {
    expect(output.find('image.jpg'));
  });
});
