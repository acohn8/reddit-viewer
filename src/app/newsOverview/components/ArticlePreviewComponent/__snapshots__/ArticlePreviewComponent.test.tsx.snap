// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlePreviewComponent should render correctly 1`] = `
<FeedEvent>
  <FeedContent>
    <Header
      as="a"
      content="New Yorker"
      href="www.newyorker.com"
      icon="newspaper outline"
      size="medium"
      subheader="3 hours ago"
      target="_blank"
    />
    <FeedSummary>
      breaking
    </FeedSummary>
    <FeedExtra
      text={true}
    >
      <p>
        a cool article2
      </p>
    </FeedExtra>
    <Image
      as="a"
      href="www.newyorker.com"
      src="image.jpg"
      target="_blank"
      ui={true}
    />
    <FeedMeta>
      <FeedLike
        as="a"
      >
        <Icon
          as="i"
          name="comments"
        />
        5 comments
      </FeedLike>
      <FeedLike
        as="a"
      >
        <Icon
          as="i"
          name="thumbs up outline"
        />
        7
      </FeedLike>
      <Icon
        as="i"
        name="bookmark"
      />
    </FeedMeta>
  </FeedContent>
</FeedEvent>
`;
